cmd/access/node_builder/staked_access_node_builder.go

```diff
@@ -105,36 +110,68 @@ func (builder *StakedAccessNodeBuilder) enqueueUnstakedNetworkInit(ctx context.C
 
 	builder.Component("unstaked network", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
 
-		// NodeID for the staked node on the unstaked network
-		// TODO: set a different node ID of the staked access node on the unstaked network
-		unstakedNodeID := builder.NodeID // currently set the same as the staked NodeID
-
-		// Networking key
-		// TODO: set a different networking key of the staked access node on the unstaked network
-		unstakedNetworkKey := builder.NetworkKey
-
-		libP2PFactory, err := builder.initLibP2PFactory(ctx, unstakedNodeID, unstakedNetworkKey)
+		libP2PFactory, err := builder.initLibP2PFactory(ctx, builder.NodeID, builder.NodeConfig.NetworkKey)
 		builder.MustNot(err)
 
-		msgValidators := unstakedNetworkMsgValidators(unstakedNodeID)
-
-		// Network Metrics
-		// for now we use the empty metrics NoopCollector till we have defined the new unstaked network metrics
-		// TODO: define new network metrics for the unstaked network
-		unstakedNetworkMetrics := metrics.NewNoopCollector()
+		msgValidators := unstakedNetworkMsgValidators(builder.NodeID)
 
-		middleware := builder.initMiddleware(unstakedNodeID, unstakedNetworkMetrics, libP2PFactory, msgValidators...)
+		middleware := builder.initMiddleware(builder.NodeID, node.Metrics.Network, libP2PFactory, msgValidators...)
 
 		// topology returns empty list since peers are not known upfront
-		top := topology.EmptyListTopology{}
+		top, err := topology.NewTopicBasedTopology(
+			builder.NodeID,
+			builder.Logger,
+			builder.State,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("could not create topology: %w", err)
+		}
+		topologyCache := topology.NewCache(builder.Logger, top)
```

Comment:
Why are we using topic based topology here? it looks like that basically just creates the topology based on interacting roles, which I'm not sure will work? Because it will also disconnect from any peers not in the list.